go/tools/gazelle/gazelle/main.go

```diff
@@ -144,23 +115,7 @@ func main() {
 	flag.Usage = usage
 	flag.Parse()
 
-	if *repoRoot == "" {
-		var err error
-		if *repoRoot, err = repo(flag.Args()); err != nil {
-			log.Fatal(err)
-		}
-	}
-	if *goPrefix == "" {
-		var err error
-		if *goPrefix, err = loadGoPrefix(*repoRoot); err != nil {
-			if !os.IsNotExist(err) {
-				log.Fatal(err)
-			}
-			log.Fatalf("-go_prefix not set and no root BUILD file found")
-		}
-	}
-
-	genericTags, err := parseBuildTags(*buildTags)
+	c, err := newConfiguration()
```

Comment:
I would change newConfiguration to take *flag.FlagSet, and pass in flag.CommandLine here so that access to the global is isolated to this point only.
Opens the door to change to a locally declared flagset in the future rather than global flags.